src/app/dim-ui/Select.tsx

```diff
@@ -70,10 +86,39 @@ export default function Select<T>({
     throw new Error('value must correspond to one of the provided options');
   }
 
+  let buttonStyle: CSSProperties | undefined;
+  let dropdownStyle: CSSProperties | undefined;
+
+  if (maxButtonWidth !== undefined) {
+    buttonStyle = {
+      maxWidth: maxButtonWidth,
+    };
+  }
+
+  if (maxDropdownWidth === 'button' && dropdownWidth === undefined && buttonRef.current) {
+    // Minus 2 because the menu has a thicker outline than the button border (2px vs 1px)
+    const width =
+      maxButtonWidth !== undefined
+        ? maxButtonWidth
+        : buttonRef.current.getBoundingClientRect().width - 2;
+    setDropdownWidth(width);
```

Comment:
Don't do this in the render. You can use an effect for this.

Reply:
Done